mitodl/open-discussions
# Update Post and Comment tables so they can be a source of truth for indexing

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #1699 
Part of #1662 

#### What's this PR do?
Adds additional fields to shadow tables so they can be a source of truth for reindexing

#### How should this be manually tested?

Test the new code:
- Apply the migrations
- Create new posts and comments and verify they get data in the new fields
- Run `./manage.py backpopulate_post_and_comment_fields`
- Verify older posts and comments have been populated with data

Test backpopulate_missing_posts:
- Delete non-article posts:
```python
from channels.models import *
Post.objects.filter(article__isnull=True).delete()
```
- Run `./manage.py backpopulate_missing_posts`
